File: 250504/신기한 타일 뒤집기/strange-flipping-tiles.cpp
#include <iostream>
#include <climits>

#define MAX_K 100000

using namespace std;

int n;
int draw[2 * MAX_K + 1];
int black, white;

int main()
{
    ios::sync_with_stdio(0);
    cin.tie(NULL);
    cout.tie(NULL);

    cin >> n;
    int current = MAX_K;

    for(int i = 0; i < n; i++) {
        int x;
        char c;
        cin >> x >> c;
        if(c == 'L') {
            while(x--) {
                draw[current] = 1;
                if(x != 0) {
                    current--;

                }
            }
        }
        else {
            while(x--) {
                draw[current] = 2;
                if(x != 0) {
                    current++;
                }
            }
        }
    }

    for(int i = 0; i < 2 * MAX_K + 1; i++) {
       if(draw[i] == 1) {
            white++;
        }
        else if(draw[i] == 2) {
            black++;
        }
    }
    cout << white << " " << black << "\n";
    return 0;
}

File: 250504/신기한 타일 뒤집기/strange-flipping-tiles.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        int[] draw = new int[200001];
        int[] black = new int[200001];
        int[] white = new int[200001];
        int blackCount = 0, whiteCount = 0;
        int current = 100000;
        for(int i = 0; i < n; i++) {
            int x = sc.nextInt();
            String s = sc.next();
            if(s.equals("L")) {
                while(x-- > 0) {
                    draw[current] = 1;
                    white[current]++;
                    if(x > 0) {
                        current--;
                    }
                }
            }
            else {
                while(x-- > 0) {
                    draw[current] = 2;
                    black[current]++;
                    if(x > 0) {
                        current++;
                    }
                }
            }
        }
        for(int i = 0; i <= 200000; i++) {
            if(draw[i] == 1) {
                whiteCount++;
            }
            else if(draw[i] == 2) {
                blackCount++;
            }
        }
        System.out.println(whiteCount + " " + blackCount);
        sc.close();
    }
}
